Assets/Countly/Plugins/CountlySDK/Services/ViewCountlyService.cs

```diff
@@ -25,198 +25,882 @@ private class ViewData
 
-                _isFirstView = false;
+                StopViewWithIDInternal(viewID, viewSegmentation);
             }
+        }
+
+        /// <summary>
+        /// Pauses a view with the given ID
+        /// </summary>
+        /// <param name="viewID">ID of the view</param>
+        public void PauseViewWithID(string viewID)
+        {
+            if (!_consentService.CheckConsentInternal(Consents.Views)) {
+                Log.Debug("[ViewCountlyService] PauseViewWithID, consent is not given, ignoring the request.");
+                return;
+            }
+
+            lock (LockObj) {
+                Log.Info("[ViewCountlyService] Calling PauseViewWithID vi[" + viewID + "]");
 
-            await Task.CompletedTask;
+                PauseViewWithIDInternal(viewID, false);
+            }
         }
 
         /// <summary>
-        /// Stop tracking a view
+        /// Resumes a view with the given ID
         /// </summary>
-        /// <param name="name of the view"></param>
-        /// <returns></returns>
-        [Obsolete("RecordCloseViewAsync(string name) is deprecated, this is going to be removed in the future.")]
-        public async Task RecordCloseViewAsync(string name)
+        /// <param name="viewID">ID of the view</param>
+        public void ResumeViewWithID(string viewID)
         {
+            if (!_consentService.CheckConsentInternal(Consents.Views)) {
+                Log.Debug("[ViewCountlyService] ResumeViewWithID, consent is not given, ignoring the request.");
+                return;
+            }
+
             lock (LockObj) {
-                Log.Info("[ViewCountlyService] RecordCloseViewAsync : name = " + name);
+                Log.Info("[ViewCountlyService] Calling ResumeViewWithID vi[" + viewID + "]");
+
+                ResumeViewWithIDInternal(viewID);
+            }
+        }
+
+        /// <summary>
+        /// Stops all views and records a segmentation if set
+        /// </summary>
+        /// <param name="viewSegmentation">view segmentation</param>
+        public void StopAllViews(Dictionary<string, object> viewSegmentation)
+        {
+            if (!_consentService.CheckConsentInternal(Consents.Views)) {
+                Log.Debug("[ViewCountlyService] StopAllViews, consent is not given, ignoring the request.");
+                return;
+            }
+
+            lock (LockObj) {
+                Log.Info("[ViewCountlyService] Calling StopAllViews sg[" + (viewSegmentation == null ? "null" : viewSegmentation.Count.ToString()) + "]");
+
+                StopAllViewsInternal(viewSegmentation);
+            }
+        }
+
+        /// <summary>
+        /// Set a segmentation to be recorded with all views
+        /// </summary>
+        /// <param name="viewSegmentation">global view segmentation</param>
+        public void SetGlobalViewSegmentation(Dictionary<string, object> viewSegmentation)
```

Comment:
For example maybe we can do dictionary validation here. Whether or not it is empty. We can eliminate bad values before passing it thought internal logic